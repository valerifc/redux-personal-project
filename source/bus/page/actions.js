// Types
import { types } from "./types";

export const pageActions = {
    setPage: (meta) => {
        return {
            type:    types.SET_PAGE,
            payload: meta,
        };
    },
};
