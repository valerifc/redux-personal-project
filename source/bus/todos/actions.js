// Types
import { types } from "./types";

export const todosActions = {
    // Sync
    fetchTodos: (todos) => {
        return {
            type:    types.FETCH_TODOS,
            payload: todos,
        };
    },
    createTodo: (todo) => {
        return {
            type:    types.CREATE_TODO,
            payload: todo,
        };
    },
    updateTodo: (todo) => {
        return {
            type:    types.UPDATE_TODO,
            payload: todo,
        };
    },
    updateTodos: (todos) => {
        return {
            type:    types.UPDATE_TODOS,
            payload: todos,
        };
    },
    updateAllTodos: (todo) => {
        return {
            type:    types.UPDATE_ALL_TODOS,
            payload: todo,
        };
    },
    removeTodo: (todoId) => {
        return {
            type:    types.REMOVE_TODO,
            payload: todoId,
        };
    },

    // Async
    fetchTodosAsync: (queryParams) => {
        return {
            type:    types.FETCH_TODOS_ASYNC,
            payload: queryParams,
        };
    },
    createTodoAsync: (message) => {
        return {
            type:    types.CREATE_TODO_ASYNC,
            payload: message,
        };
    },
    updateTodoAsync: (todo) => {
        return {
            type:    types.UPDATE_TODO_ASYNC,
            payload: todo,
        };
    },
    removeTodoAsync: (todoId) => {
        return {
            type:    types.REMOVE_TODO_ASYNC,
            payload: todoId,
        };
    },
};
