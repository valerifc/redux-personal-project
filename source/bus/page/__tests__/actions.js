// Actions
import { pageActions } from "../actions";

describe('page actions:', () => {
    test('startFetching', () => {
        expect(pageActions.setPage(__.page)).toMatchSnapshot();
    });
});
