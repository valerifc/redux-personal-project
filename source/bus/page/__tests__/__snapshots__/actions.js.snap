// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`page actions: startFetching 1`] = `
Object {
  "payload": Object {
    "allCompleted": true,
    "newTodoText": "TEST_TEXT",
    "page": 2,
    "search": "test_search",
    "size": 10,
    "total": 17,
  },
  "type": "SET_PAGE",
}
`;
